CRM/Price/BAO/LineItem.php

```diff
@@ -641,22 +633,19 @@ public static function changeFeeSelections(
     // fetch submitted LineItems from input params and feeBlock information
     $submittedLineItems = $lineItemObj->getSubmittedLineItems($params, $feeBlock);
 
-    // retrieve the submitted price field value IDs from $submittedLineItems array keys
-    $submittedPriceFieldValueIDs = empty($submittedLineItems) ? array() : array_keys($submittedLineItems);
-
     $requiredChanges = $lineItemObj->getLineItemsToAlter($submittedLineItems, $entityID, $entity);
 
-    // cancel previous line item
-    $additionalWhereClause = empty($submittedPriceFieldValueIDs) ? NULL : sprintf("price_field_value_id NOT IN (%s)", implode(', ', $submittedPriceFieldValueIDs));
-    $lineItemObj->cancelLineItems($entityID, $entityTable, $additionalWhereClause);
-
-    // get financial information that need to be recorded on basis on submitted price field value IDs
-    $financialItemsArray = $lineItemObj->getReverseFinancialItemsToRecord(
-      $entityID,
-      $entityTable,
-      $contributionId,
-      $submittedPriceFieldValueIDs
-    );
+    $financialItemsArray = array();
+    if (!empty($requiredChanges['line_items_to_cancel']) || !empty($requiredChanges['line_items_to_update'])) {
```

Comment:
@eileenmcnaughton I am getting a issue re text field qty change for adding the IF condition here. Here's the steps to replicate:
1. Do a pending Event registration with text price option 1x10$ 
2. Then change fee of text qty, say 2 and then save 
Then on financial item table, we should expect 3 records:
 - First is the old financial item ($10, line_item, 111) as financial_item(amount, entity_table, entity_id)
 -  Second is the reversal item (-$10, line_item, 111)
 -  Third is the new item ($20, line_item, 111)

On top of this patch, I am not getting the reverse entry that is the 2nd record. This is the fix https://gist.github.com/monishdeb/c55891e6795f5f4da73f2898ab780baa

Reply:
@monishdeb I'm not super keen about changing code to cover more cases without adding more tests. However, your change affects code within this PR and doesn't cause the existing unit tests to fail per se - so I've changed it.

Oddly the existing unit tests DID start to fail because this line of code started to be hit

CRM_Contribute_BAO_Contribution::getFinancialAccountId()

And that function seems not to exist at all! Either before or after this. I have simply removed the code that calls that function for now - since it's clearly not working. But I think it should be re-introduced with appropriate unit tests